src/form-engine.test.tsx

```diff
@@ -1047,6 +1098,86 @@ describe('Form engine component', () => {
     });
   });
 
+  describe('Diagnisis field', () => {
+    it('should test addition of a diagnosis', async () => {
+      await act(async () => {
+        renderForm(null, diagnosisForm);
+      });
+      const addButtons = screen.getAllByRole('button', { name: 'Add' });
+      expect(addButtons.length).toBeGreaterThan(0);
+      screen.debug(addButtons);
+
+      await user.click(addButtons[0]);
+
+      await waitFor(() => {
+        expect(screen.getAllByRole('combobox', { name: /^test diagnosis 1$/i }).length).toEqual(2);
+      });
+
+      expect(screen.getByRole('button', { name: /Remove/i })).toBeInTheDocument();
+    });
+
+    it('should render diagnosis field', async () => {
+      await act(async () => {
+        renderForm(null, diagnosisForm);
+      });
+      const initialDiagnosis = screen.getAllByRole('combobox', { name: /test diagnosis 1|test diagnosis 2/i });
+      expect(initialDiagnosis.length).toBe(2);
+    });
+
+    it('should save diagnosis field on form submission', async () => {
+      await act(async () => {
+        renderForm(null, diagnosisForm);
+      });
+      const saveEncounterMock = jest.spyOn(api, 'saveEncounter');
+
+      const combobox = await screen.findByRole('combobox', { name: /test diagnosis 1/i });
+      expect(combobox).toBeInTheDocument();
+
+      await userEvent.click(combobox);
+      await waitFor(() => {
+        expect(screen.getByText('stage 1')).toBeInTheDocument();
+        expect(screen.getByText('stage 2')).toBeInTheDocument();
+      });
+      await user.click(screen.getByText('stage 1'));
+      await user.click(screen.getByRole('button', { name: /save/i }));
+      expect(saveEncounterMock).toHaveBeenCalledTimes(1);
+      const [_, encounter] = saveEncounterMock.mock.calls[0];
+      expect(encounter.diagnoses.length).toBe(1);
+      expect(encounter.diagnoses[0]).toEqual({
+        patient: '8673ee4f-e2ab-4077-ba55-4980f408773e',
+        condition: null,
+        diagnosis: {
+          coded: 'stage-1-uuid',
+        },
+        certainty: 'CONFIRMED',
+        rank: 1,
+        formFieldPath: `rfe-forms-diagnosis1`,
+        formFieldNamespace: 'rfe-forms',
+      });
+    });
```

Comment:
Can you also add a unit test validating that the "repeating container" is properly constructed in edit mode?